src/pydata_sphinx_theme/__init__.py

```diff
@@ -49,33 +50,61 @@ def update_config(app, env):
         )
 
     # Add an analytics ID to the site if provided
-    # Currently only supports the two types of Google Analytics id.
+
+    # deprecated options for Google Analytics
     gid = theme_options.get("google_analytics_id")
     if gid:
-        # In this case it is "new-style" google analytics
-        if "G-" in gid:
-            gid_js_path = f"https://www.googletagmanager.com/gtag/js?id={gid}"
-            gid_script = f"""
-                window.dataLayer = window.dataLayer || [];
-                function gtag(){{ dataLayer.push(arguments); }}
-                gtag('js', new Date());
-                gtag('config', '{gid}');
-            """
-        # In this case it is "old-style" google analytics
+        msg = (
+            "'google_analytics_id' is deprecated and will be removed in "
+            "version 0.11, please refer to the documentation "
+            "and use 'analytics' instead."
+        )
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+        if theme_options.get("analytics"):
+            theme_options["analytics"].update({"google_analytics_id": gid})
         else:
-            gid_js_path = "https://www.google-analytics.com/analytics.js"
-            gid_script = f"""
-                window.ga = window.ga || function () {{
-                    (ga.q = ga.q || []).push(arguments) }};
-                ga.l = +new Date;
-                ga('create', '{gid}', 'auto');
-                ga('set', 'anonymizeIp', true);
-                ga('send', 'pageview');
-            """
+            theme_options["analytics"] = {"google_analytics_id": gid}
 
-        # Link the JS files
-        app.add_js_file(gid_js_path, loading_method="async")
-        app.add_js_file(None, body=gid_script)
+    analytics = theme_options.get("analytics")
```

Comment:
why not define analytics above with `analytics = theme_options.get("analytics", {})`, then you could just call the `.update` command above like `analytics.update({"google_analytics_id": gid})` and wouldn't need the conditional

Reply:
I will have a look